[WIP] Add compatibility with @material-ui/core v4

leocabeza opened the pull request:
Related: #275

I did my best to update dependencies and tests to make this work for @material-ui/core v4.

Work pending (I need feedback). For the following tests, I commented out two lines:

- adds the input on blur with blurBehavior set to add with timeout (1)
- adds the input on blur with blurBehavior set to add without timeout (2)

They are related to _setTimeout_ being called, I believe this is due to implementation details of @material-ui/core v4.

For (1), _setTimeout_ is called five times (instead of one), and for (2), it's called two times (instead of zero).

Any other feedback is also appreciated.

cc @leMaik

leMaik commented:
Thank you very much for your time and for updating this! :clap: I'll look into this later this weekend.

leMaik commented:
@leocabeza Again, thanks for the updating effort. I rebased your branch against the `master` branch and now, for some reason, the `defaultValue` tests fail. :thinking: Would you like to look into that?

leMaik commented on package.json:
```diff
@@ -44,29 +44,29 @@
-    "@material-ui/core": "^1.0.0 || ^3.1.0",
-    "react": "^16.3.0",
-    "react-dom": "^16.3.0"
+    "@material-ui/core": "^1.0.0 || ^3.1.0 || ^4.0.0",
```
It would be sufficient to only support `^4.0.0` here. People can use v1.0.0 of this component for compatibility with Material-UI 3.x

leMaik commented on src/ChipInput.spec.js:
```diff
@@ -512,12 +512,12 @@ describe('blurBehavior modes', () => {
 
     expect(tree.find('input').getDOMNode().value).toBe('')
-    expect(setTimeout).toHaveBeenCalledTimes(0)
+    /* expect(setTimeout).toHaveBeenCalledTimes(0) */
```
I'm not sure if we even need to test how often `setTimeout` was called. If there was a timeout before adding the chip, the next `expect` should fail because we didn't run it in the meantime.

leocabeza replied:
So, should we delete this line then?

leocabeza commented:
> @leocabeza Again, thanks for the updating effort. I rebased your branch against the `master` branch and now, for some reason, the `defaultValue` tests fail.  Would you like to look into that?

Yeah, I'll take a look

leocabeza commented:
- Updated failing tests
- Removed the expect call on _setTimeout_ on the tests
- Support only material-ui v4 on package.json peerDependencies

Let me know if this is good enough or a change is needed @leMaik

leMaik commented:
@leocabeza Thank you for the update! :+1: I'll merge this into a `next` branch. Let's continue the work from there.
